standard/variables.md

```diff
@@ -987,3 +987,175 @@ variable_reference
 ## 9.6 Atomicity of variable references
 
 Reads and writes of the following data types shall be atomic: `bool`, `char`, `byte`, `sbyte`, `short`, `ushort`, `uint`, `int`, `float`, and reference types. In addition, reads and writes of enum types with an underlying type in the previous list shall also be atomic. Reads and writes of other types, including `long`, `ulong`, `double`, and `decimal`, as well as user-defined types, need not be atomic. Aside from the library functions designed for that purpose, there is no guarantee of atomic read-modify-write, such as in the case of increment or decrement.
+
+## §ref-span-safety Reference variables and returns
+
+### §ref-span-safety-general General
+
+A ***reference variable*** is a variable that refers to another variable, called the ***referent***. Therefore, a reference variable does not store its value. Instead, a reference variable refers to its referent. Thus, the value of a reference variable is always the same as its referent variable. A reference variable is a local variable declared with the `ref` modifier.
```

Comment:
This is a bit torturous, *"a reference variable does not store its value"* – it does, what it doesn't store is the *referent's value*. I understand what is being attempted – to not talk about the *value* stored in the ref variable as much as possible, except we have to as we can reassign that value so the variable refers to a different referent…

There is a comment below by @jaredpar, its hidden now so here's an extract: 

> Slight tangent ... if you think about ref as a type vs. an annotation on a location it often makes the rules around ref safety easier to grok (at least to me). Consider that if we had ref<int> instead of ref int then having a single term becomes completely natural here as everything is just a value.

What we're being fuzzy over is that it is a type, a reference variable holds a value of type "*reference to a variable*" – echoes of Algol-68 `ref ref int x := ref int y` ;-), but that languages attempt to make all this clear confused a *lot* of people…

Anyway I'll attempt a bit of wordsmithing here but really we need to consider how we're futzing this in general:

```suggestion
A ***reference variable*** is a variable that refers to another variable, called the ***referent***. A reference variable does not store the value of its referent. When a reference variable is used where a value is required its referent's value is returned; similarly when a reference variable is the target of an assignment it is the referent which is assigned to. The variable to which a reference variable refers, i.e. its referent, can be changed using a ref assignment (`= ref`) [insert x-ref]. A reference variable is a local variable declared with the `ref` modifier.
```
I can't say I'm totally happy with that, maybe someone else can improve it, but I think its a bit clearer.